adapters/facebook_test.go

```diff
@@ -285,6 +307,220 @@ func TestFacebookBasicResponse(t *testing.T) {
+		publisherID: "987654321",
+		tags:        make([]tagInfo, 1),
+		deviceIP:    "25.91.96.36",
+		deviceUA:    "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_4) AppleWebKit/603.1.30 (KHTML, like Gecko) Version/10.1 Safari/603.1.30",
+		buyerUID:    "need-an-actual-fb-id",
+	}
+	fbdata.tags[0] = tagInfo{
+		code:        "first-tag",
+		placementID: fmt.Sprintf("%s_999998888", fbdata.publisherID),
+		bid:         1.67,
+		W:           300,
+		H:           250,
+	}
+
+	conf := *DefaultHTTPAdapterConfig
+	an := NewFacebookAdapter(&conf, fmt.Sprintf("%d", fbdata.partnerID), "localhost")
+	an.URI = server.URL
+	an.nonSecureUri = server.URL
+
+	pbin := pbs.PBSRequest{
+		AdUnits: make([]pbs.AdUnit, 2),
+	}
+	for i, tag := range fbdata.tags {
+		pbin.AdUnits[i] = pbs.AdUnit{
+			Code:       tag.code,
+			MediaTypes: []string{"BANNER"},
+			Sizes: []openrtb.Format{
+				{
+					W: tag.W,
+					H: tag.H,
+				},
+			},
+			Bids: []pbs.Bids{
+				pbs.Bids{
+					BidderCode: "audienceNetwork",
+					BidID:      fmt.Sprintf("random-id-from-pbjs-%d", i),
+					Params:     json.RawMessage(fmt.Sprintf("{\"placementId\": \"%s\"}", tag.placementID)),
+				},
+			},
+			Instl: 1,
+		}
+	}
+
+	body := new(bytes.Buffer)
+	err := json.NewEncoder(body).Encode(pbin)
+	if err != nil {
+		t.Fatalf("Json encoding failed: %v", err)
+	}
+
+	fmt.Println("body", body)
+
+	req := httptest.NewRequest("POST", server.URL, body)
+	req.Header.Add("Referer", fbdata.page)
+	req.Header.Add("User-Agent", fbdata.deviceUA)
+	req.Header.Add("X-Real-IP", fbdata.deviceIP)
+
+	pc := pbs.ParsePBSCookieFromRequest(req)
+	pc.TrySync("audienceNetwork", fbdata.buyerUID)
+	fakewriter := httptest.NewRecorder()
+	pc.SetCookieOnResponse(fakewriter, "")
+	req.Header.Add("Cookie", fakewriter.Header().Get("Set-Cookie"))
+
+	cacheClient, _ := dummycache.New()
+	pbReq, err := pbs.ParsePBSRequest(req, cacheClient)
+
+	ctx := context.TODO()
+	bids, err := an.Call(ctx, pbReq, pbReq.Bidders[0])
+	for _, bid := range bids {
+		matched := false
+		for _, tag := range fbdata.tags {
+			if bid.AdUnitCode == tag.code {
+				matched = true
+				if bid.BidderCode != "audienceNetwork" {
+					t.Errorf("Incorrect BidderCode '%s'", bid.BidderCode)
+				}
+				if bid.Price != tag.bid {
+					t.Errorf("Incorrect bid price '%.2f' expected '%.2f'", bid.Price, tag.bid)
+				}
+				if bid.Width != tag.W || bid.Height != tag.H {
+					t.Errorf("Incorrect bid size %dx%d, expected %dx%d", bid.Width, bid.Height, tag.W, tag.H)
+				}
+				if bid.Adm != tag.content {
+					t.Errorf("Incorrect bid markup '%s' expected '%s'", bid.Adm, tag.content)
+				}
+			}
+		}
+		if !matched {
+			t.Errorf("Received bid for unknown ad unit '%s'", bid.AdUnitCode)
+		}
+	}
+}
+
+func TestFacebookBannerRequestWithSupportedSizes(t *testing.T) {
```

Comment:
There's a lot of copy/pasted code blocks between these two methods... can we group them into helper methods?

Duplicated code makes changes harder.